src/main/java/com/synopsys/integration/alert/channel/hipchat/descriptor/HipChatDistributionUIConfig.java

```diff
@@ -28,25 +28,21 @@
 
 import org.springframework.stereotype.Component;
 
-import com.synopsys.integration.alert.channel.hipchat.HipChatChannel;
 import com.synopsys.integration.alert.common.descriptor.config.field.CheckboxConfigField;
 import com.synopsys.integration.alert.common.descriptor.config.field.ConfigField;
 import com.synopsys.integration.alert.common.descriptor.config.field.NumberConfigField;
 import com.synopsys.integration.alert.common.descriptor.config.field.SelectConfigField;
-import com.synopsys.integration.alert.common.descriptor.config.ui.DescriptorMetadata;
 import com.synopsys.integration.alert.common.descriptor.config.ui.UIConfig;
-import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
-import com.synopsys.integration.alert.common.enumeration.DescriptorType;
 
 @Component
 public class HipChatDistributionUIConfig extends UIConfig {
 
-    @Override
-    public DescriptorMetadata generateDescriptorMetadata() {
-        return new DescriptorMetadata("HipChat", "hipchat", HipChatChannel.COMPONENT_NAME, DescriptorType.CHANNEL, ConfigContextEnum.DISTRIBUTION, "comments", setupFields());
+    public HipChatDistributionUIConfig() {
+        super("HipChat", "hipchat", "comments");
```

Comment:
Encapsulate lowercase hipchat and comments strings into static final strings in the descriptor class since they are the URL and font-awesome icon name.